crates/data_components/src/oracle/convert.rs

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::sync::Arc;
+
+use crate::oracle::FailedToConvertBigDecimalToI128Snafu;
+use crate::oracle::FailedToParseBigDecimalSnafu;
+use arrow::{
+    array::{
+        ArrayBuilder, ArrayRef, BooleanBuilder, Decimal128Builder, Float32Builder, Float64Builder,
+        LargeStringBuilder, RecordBatch, RecordBatchOptions, StringBuilder, make_builder,
+    },
+    datatypes::{DataType, SchemaRef},
+};
+use bigdecimal::BigDecimal;
+use oracle::Row;
+use snafu::OptionExt;
+use snafu::ResultExt;
+
+/// Oracle Built-in Data Types
+/// `<https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Data-Types.html#GUID-7B72E154-677A-4342-A1EA-C74C1EA928E6>`
+pub(crate) fn map_oracle_type_to_arrow_type(
```

Comment:
This is initial implementation that will be extended and revised next, including type conversion tests